### ISSUE TITLE
Can't install extensions: "To add this item to Chrome, please update your browser"

### ISSUE THREAD

**Issue by AntiSol** [brave/brave-browser#40377]
### Description

I cannot install extensions in brave. On every desktop machine I've ever used brave on, when trying to install extensions, I see the message "To add this item to Chrome, please update your browser.", and the "add to (brave|chrome)" button is disabled.

This has been an issue for years. I used to be able to work around this by going to a legacy version of the chrome web store at https://chrome.google.com/webstore/old, but this site a) advises that it will be discontinued soon and b) is now redirecting me to the new chrome store whenever I try to use it to install an extension

### Steps to reproduce

1. install brave from your distro's repos
2. go to tools -> extensions
3. search for an extension, click on one
4. observe the inability to install extensions and the requirement to upgrade to a newer version of chrome

### Actual result

![image](https://github.com/user-attachments/assets/0dfabac8-bebd-45d3-8c2d-be76699a01c9)

### Expected result

ability to install extensions

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

Brave	1.36.119 Chromium: 99.0.4844.83 (Official Build) (64-bit) 

### Channel information

- [X] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [X] with Brave Shields disabled
- [X] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by rebron** [comment]
@AntiSol Can you update to the latest version of Brave? What OS are you using?
Our latest is 1.68.137.

**Comment by AntiSol** [comment]
I'm using the latest version available in my distro's repositories. The one running 1.36.119 is on manjaro but I also have an ubuntu machine which will likely be using a slightly older version than that.

I don't install or run software from untrusted sources, so updating to the latest version is nontrivial.

I think brave should probably just spoof its user agent (or whatever is being used for detection) to make it look like the latest version on sites like this.

**Comment by AntiSol** [comment]
just curious why this has the "needs more info" label. What info has been requested?

**Comment by ThumpieBunnyEve** [comment]
Issue remains open not quite a year later. Update requested.

**Comment by bsclifton** [comment]
Hi folks - sorry, one piece of information asked for was OS. So this is Linux

@AntiSol @ThumpieBunnyEve did you install using the instructions on https://brave.com/linux?

If not, please let us know how it was installed (ex: snap, flatpak, etc)

**Comment by AntiSol** [comment]
yes

**Comment by bsclifton** [comment]
@AntiSol yes - you shared one machine is manjaro and other is ubuntu.

But do you have more information? Like which version of it? For ubuntu, you can run `cat /etc/lsb-release` for example

**Comment by AntiSol** [comment]
@bsclifton as mentioned, this has been a thing for years. On multiple OS's and versions. I'm not sure that providing the OS versions I'm using right now would be particularly helpful because it's been like this on literally every operating system I've ever tried and with every brave version I've ever used. It seems to me that the specific OS version is unlikely to be very relevant, and that it's much more likely to be tied to the brave version (which I provided in my initial report). 

But I'm sure I could be convinced to provide more specific OS info, if you'd care to to elaborate on why you think the OS version is important - does brave interact with the chrome web store differently depending on your OS version? If so, why does it do that? (that seems like really strange behaviour to me) 

Are you having trouble replicating it on manjaro / ubuntu? That would be strange, since as mentioned it's been super easy for me to replicate on literally every OS and brave version I've ever tried on. 

Which OS and brave versions did you try? My first suggestion would be to try that specific version of brave I reported rather than a different OS.

Very happy to help further if you get back to me with answers to my questions 🙂

**Comment by bsclifton** [comment]
@AntiSol sorry for long response time

I've never had a problem on Ubuntu - I haven't used Manjaro before. 

cc: @fmarier @fallaciousreasoning in case they have an idea

**Comment by fmarier** [comment]
I've also never had a problem with Ubuntu or Debian. I'm not aware of other users reporting these problems.

According to https://manjaristas.org/branch_compare?q=brave, Brave 1.77 is available in Manjaro and so you should make sure you are running that version. I would expect incompatibilities between the Chrome Web Store and very old versions of Brave, but not with the latest version.

**Comment by fallaciousreasoning** [comment]
Yeah, just seconding this - I would expect old versions of Brave to have a problem installing new extensions. For reference 1.36.119 was released ~3 years ago and is based on Chromium version 99.0.4844.83.

If you're running a version that old you should be pretty careful as there are a lot of security patches between then and now!

If you follow the install instructions for Linux again to get the latest version is it still broken in Ubuntu? I'm running latest Nightly, Beta & Release on Ubuntu 24.10 and all work fine for installing extensions.

If you don't want to add the Brave ppa you could try installing the snap from the `App Center` but I don't think its maintained by us (seems to be up to date ish though, on 1.77).

**Comment by fmarier** [comment]
> Could I ask for some more information as to exactly why an old version would have a problem installing new extensions? Is it based on the user agent (I don't think so, I tried messing with my user agent), is there some "webextensions version" signifier somewhere? What exactly is the chrome store looking at to determine whether it shows the 'install' button or the 'upgrade' message?

That's a good question. I'm not sure why Google blocks old clients. It could be because of incompatibilities with recent extensions (or the Chrome Web Store itself), or it could be for security reasons.

But whatever the reason is, it's good that it caused you to notice you had a very out of date browser and fix the underlying update problem. Browsers change a lot and many of the updates contain security fixes or improvements.

**Comment by AntiSol** [comment]
My question was more about the how than the reasoning behind it. 

You said "I would expect incompatibilities between the Chrome Web Store and very old versions of Brave, but not with the latest version". 

I am asking you on what technical basis would you expect this, and what is the mechanism behind the block if not the user agent.

I'm also still curious as to where the line between "old" and not is.


(btw I happen to know for a fact that there weren't actually any known security issues which were relevant to my situation and fixed in this upgrade, so the vague "for security reasons" response that people love to trot out by reflex doesn't hold water here)

**Comment by fmarier** [comment]
> I am asking you on what technical basis would you expect this, and what is the mechanism behind the block if not the user agent.

I don't know the technical reasons (or mechanism) for why the Chrome Web Store blocks old clients. What I was expressing is that I'm not surprised they don't extend their backwards compatibility forever and put a limit on what versions are supported.

If you are interested in these technical details, I would suggest you ask the [Chrome engineers](https://source.chromium.org/chromium/chromium/src/+/main:extensions/OWNERS) who maintain this code.

**Comment by AntiSol** [comment]
thanks!
